src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMSourceHelper.java

```diff
@@ -0,0 +1,131 @@
+
+    @NonNull
+    String repo;
+
+    @NonNull
+    String owner;
+
+    @NonNull
+    String repoName;
+
+    @NonNull
+    URL url;
+
+    private static final Logger LOGGER = Logger.getLogger(GitHubSCMSourceHelper.class.getName());
+
+    @SuppressFBWarnings
+    private GitHubSCMSourceHelper(){}
+
+    @NonNull
+    public static GitHubSCMSourceHelper build(GitHubSCMSource source) {
+        GitHubSCMSourceHelper helper = new GitHubSCMSourceHelper();
+        if (source == null) return helper;
+        try {
+            helper.apiURI = getUri(source.rawUrl, source.getApiUri());
+            helper.repo = getRepoFullName(source);
+            String[] split = helper.repo.split("/");
+            if (split != null && split.length == 2) {
+                helper.owner = split[0];
+                helper.repoName = split[1];
+            }
+            helper.url = getUrl(source);
+
+        } catch (MalformedURLException e) {
+            LOGGER.log(Level.WARNING, "Malformed repository URL {0}", source.rawUrl);
+        }
+        return helper;
+
+    }
+    @NonNull
+    private static URL getUrl(GitHubSCMSource source) throws MalformedURLException {
+        if(StringUtils.isNotBlank(source.getRawUrl()) ){
+            return new URL (removeEnd(source.getRawUrl(),".git"));
+        }else{
+            if( StringUtils.isBlank(source.getApiUri())){
+                return new URL("https://github.com/" + source.repoOwner+"/"+source.repository);
+            }else{
+                String baseURL = removeEnd(source.getApiUri(), "/api/v3");
+                return new URL(baseURL+ "/" + source.repoOwner+"/"+source.repository);
+            }
+
+        }
+    }
+
+    @NonNull
+    private static String getUri(String rawUrl, String apiUri) {
+        //RawURL
+        if (StringUtils.isNotBlank(rawUrl)) {
+            return getRawUrlUri(rawUrl);
+        } else if (StringUtils.isBlank(apiUri)) { //GitHub Repo
+            return "https://api.github.com";
+        }else {
+            // GitHub Enterprise or rawUrl
+            return apiUri;
+        }
+    }
+
+    @NonNull
+    private static String getRawUrlUri(String rawUrl) {
+        String uri = rawUrl;
+        URL url;
+        try {
+            url = new URL(rawUrl);
+            if ("github.com".equals(url.getHost())) {
+                uri = "https://api." + url.getHost();
+            } else {
+                String port = "";
+                boolean httpOk = "http".equals(url.getProtocol()) && url.getPort() == 80;
+                boolean httpsOk = "https".equals(url.getProtocol()) && url.getPort() == 443;
+                if (!httpOk && !httpsOk && url.getPort()!= -1){
+                    port = ":"+url.getPort();
+                }
+                uri = url.getProtocol()+"://" + url.getHost() + port +  "/api/v3";
+            }
+        } catch (MalformedURLException e) {
+            LOGGER.log(Level.WARNING, "Malformed repository URL {0}", rawUrl);
```

Comment:
Does it make sense to swallow this exception, or should it be thrown out of the method so callers see that there is a problem? Same with the other methods in the class.